test/extensions/filters/http/ext_proc/filter_test.cc

```diff
@@ -2261,6 +2272,90 @@ TEST_F(HttpFilterTest, FailOnInvalidHeaderMutations) {
   filter_->onDestroy();
 }
 
+class HttpFilter2Test : public HttpFilterTest, public Http::HttpConnectionManagerImplMixin {};
+
+// Test proves that when onData(data, end_stream) is called before headers response is returned,
+// ext_proc filter will make the data buffered in ActiveStream buffer without triggering a buffer
+// over high watermark call, which ends in an 413 error return on request path, or a 500 error on
+// response path.
```

Comment:
I think this just test request case? Should we test response case?

Reply:
er, I can add one if needed, but I think it's the same shared code path that has been covered by this test already: 
common process after De/encodeData, and the ext_proc onData() change here.

Reply:
It might not always be the case and it might be good to avoid a bug sneaking into one side of this?  we could augment this existing test to also test the response

Reply:
done, added a new test.